test/framework/clustervalidator.go

```diff
@@ -0,0 +1,182 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+)
+
+type clusterValidation = func(ctx context.Context, validateOpts clusterOpts) error
+
+type retriableValidation struct {
+	validation    clusterValidation // the validation to run against the cluster.
+	backoffPeriod time.Duration     // the duration to wait another validation attempt.
+	maxRetries    int               // the maximum number of retries to validate.
+}
+
+// ClusterValidator is responsible for checking if a cluster is valid against the spec that is provided.
+type ClusterValidator struct {
+	ClusterOpts clusterOpts
+	validations []retriableValidation
+}
+
+// WithValidation registers a validation to the ClusterValidator that will be run when Validate is called.
+func (c *ClusterValidator) WithValidation(validation clusterValidation, backoffPeriod time.Duration, maxRetries int) {
+	c.validations = append(c.validations, retriableValidation{validation, backoffPeriod, maxRetries})
+}
+
+// WithWorkloadClusterValidations registers a validation set for a management cluster.
+func (c *ClusterValidator) WithWorkloadClusterValidations() {
+	c.WithValidation(validateClusterReady, 5*time.Second, 60)
+	c.WithValidation(validateControlPlaneNodes, 5*time.Second, 60)
```

Comment:
Are we going to add a check for worker nodes?

Reply:
Yes, we will! Currently working on checking for the existence of EKSA objects first, but that is the plan